gnocchi/rest/aggregates/api.py

```diff
@@ -365,20 +391,56 @@ def get_measures(self, groups):
                         [resource], self.references, self.body["operations"],
                         start, stop, self.granularity, self.needed_overlap,
                         self.fill, details=self.details)
+
+                    LOG.debug(
+                        "[%s] measure found for resource [%s], operations "
+                        "[%s], start [%s], stop [%s], granularity [%s], "
+                        "need_overlap [%s], fill [%s], details [%s], and "
+                        "references [%s].", measure, resource,
+                        self.body["operations"], start, stop, self.granularity,
+                        self.needed_overlap, self.fill, self.details,
+                        self.references)
+
                     group.add_measures(measure, date_map, start, stop,
                                        self.details)
-                except indexer.NoSuchMetric:
+                except indexer.NoSuchMetric as e:
+                    LOG.debug("No measure found for resource [%s], operations "
+                              "[%s], start [%s], stop [%s], granularity [%s], "
+                              "need_overlap [%s], fill [%s], details [%s], "
+                              "and references [%s]. Error: [%s].", resource,
+                              self.body["operations"], start, stop,
+                              self.granularity, self.needed_overlap,
+                              self.fill, self.details, self.references, e)
                     continue
 
         self.truncate_measures()
         self.sum_sequential_group_metrics(groups)
         return groups
 
     def get_date_map(self, group):
-        if 'id' in group.group_key:
-            return self.measures_date_map.setdefault(group.group_key['id'], {})
+        """This method returns the map used to control the groupby elements.
+
+        If the entry does not exist, an empty map is returned. The rest of the
+        code handles/works by adding entries to the map returned by this method
+        """
+        data_map_key = ""
+        all_keys = list(group.group_key.keys())
+        if not all_keys:
+            LOG.error("A group without keys for [%s] should not be possible.",
+                      group)
 
         return self.measures_date_map
+        all_keys.sort()
+        for key in all_keys:
+            data_map_key += "%s=%s," % (key, group.group_key[key])
+
+        data_map_key = data_map_key[0:len(data_map_key) - 1]
+
+        date_map_to_return = self.measures_date_map.setdefault(data_map_key, {})
+        LOG.debug("Date map [%s] found for key [%s].",
+                  date_map_to_return, data_map_key)
+
+        return date_map_to_return
```

Comment:
this a copy/paste issue? everything after return in 432 will never run/

Reply:
Yes, it is. Nice catch